MIGRATING-0.6-0.7.md

```diff
@@ -396,4 +288,259 @@ Note that the table above shows the new fonts' `ascii` variants only. Some new f
+
+Note that `Dimensions` and `OriginDimensions` should be imported from `embedded-graphics-core`, not `embedded-graphics`. See the [relevant section](#the-embedded-graphics-core-crate) for more details.
+
+For example, the `SSD1306` driver using the on/off `BinaryColor` would change as follows:
+
+```diff
+- use crate::{
+-     drawable::Pixel,
+-     geometry::Size,
+-     pixelcolor::{PixelColor, BinaryColor},
+-     DrawTarget,
+- };
+-
+- impl DrawTarget<BinaryColor> for Ssd1306 {
+-     type Error = core::convert::Infallible;
+-
+-     fn draw_pixel(&mut self, pixel: Pixel<BinaryColor>) -> Result<(), Self::Error> {
+-         // ...
+-
+-         Ok(())
+-     }
+-
+-     fn size(&self) -> Size {
+-         // ...
+-     }
+- }
++ use embedded_graphics_core::{
++     draw_target::DrawTarget,
++     geometry::{OriginDimensions, Size},
++     pixelcolor::{PixelColor, BinaryColor},
++     Pixel,
++ };
++
++ DrawTarget for Ssd1306 {
++     type Color = BinaryColor;
++     type Error = core::convert::Infallible;
++
++     fn draw_iter<I>(&mut self, pixels: I) -> Result<(), Self::Error>
++     where
++         I: IntoIterator<Item = Pixel<Self::Color>>,
++     {
++         // ...
++
++         Ok(())
++     }
++ }
++
++ impl OriginDimensions for Ssd1306 {
++     fn size(&self) -> Size {
++         // ...
++     }
++ }
+```
+
+### Method changes
+
+All `draw_*` methods to draw specific primitives (`draw_circle`, `draw_triangle`, etc) have been removed. These methods were hard to implement correctly and consistently between different drivers. The new lower level draw methods are easier to implement and still improve performance over pixel by pixel drawing.
+
+- `draw_iter`
+
+  Draws individual pixels to the display without a defined order. This is the only required method in this trait, however will likely be the slowest pixel drawing implementation as it cannot take advantage of hardware accelerated features (e.g. filling a given area with a solid color with `fill_solid`).
+
+- `fill_contiguous`
+
+  Fills a given area with an iterator providing a contiguous stream of pixel colors. This may be used to efficiently draw an image or other non-transparent item to the display. The given pixel iterator can be assumed to be contiguous, iterating from top to bottom, each row left to right. This assumption potentially allows more efficient streaming of pixel data to a display.
+
+- `fill_solid`
+
+  Fills a given area with a solid color.
+
+- `clear`
+
+  Fills the entire display with a solid color.
+
+These methods aim to be more compatible with hardware-accelerated drawing commands. Where possible, embedded-graphics drawables will use `fill_contiguous` and `fill_solid` to improve performance, however may fall back to `draw_iter` by default.
+
+To reduce duplication, please search the `DrawTarget` documentation on <https://docs.rs/embedded-graphics-core> for more details on the usage and arguments of the above methods.
+
+## For crates that handle images
```

Comment:
I've clearly struggled with this too. Any of these?

- Image format support crates
- Image decoding crates
- Image crates

I'm not keen on any of them but it might inspire something better from your side.

Reply:
Out of these three I think I would use "Image format support creates". It's long, but is the most accurate description. Should we just go with that for now and change it later if we find a better description?

Reply:
There's a golden phrase out there for sure, but yeah for now I've gone with the verbose option.